modules/setting/setting.go

```diff
@@ -136,7 +136,7 @@ func forcePathSeparator(path string) {
 // This check is ignored under Windows since SSH remote login is not the main
 // method to login on Windows.
 func IsRunUserMatchCurrentUser(runUser string) (string, bool) {
-	if IsWindows || SSH.StartBuiltinServer {
+	if IsWindows || SSH.StartBuiltinServer || ignoreCheckRunUser {
```

Comment:
I do not think it is right to add this check here.

Instead, the callers should decide whether to call "IsRunUserMatchCurrentUser"